test/unit/resources/aws_config_delivery_channel_test.rb

```diff
@@ -0,0 +1,128 @@
+#=============================================================================#
+
+class AwsConfigDeliveryChannelRecallTest < Minitest::Test
+  def setup
+    AwsConfigDeliveryChannel::BackendFactory.select(AwsMDCSB::Basic)
+  end
+
+  def test_search_hit_by_default
+    assert AwsConfigDeliveryChannel.new.exists?
+  end
+
+  def test_search_hit_via_scalar
+    assert AwsConfigDeliveryChannel.new('default').exists?
+  end
+
+  def test_search_hit_via_hash
+    assert AwsConfigDeliveryChannel.new(channel_name: 'default').exists?
+  end
+
+  def test_search_miss_is_not_an_exception
+    refute AwsConfigDeliveryChannel.new(channel_name: 'NonExistentDeliveryChannel').exists?
+  end
+end
+
+#=============================================================================#
+#                               properties
+#=============================================================================#
+
+class AwsConfigDeliveryChannelPropertiesTest < Minitest::Test
+  def setup
+    AwsConfigDeliveryChannel::BackendFactory.select(AwsMDCSB::Basic)
+  end
+
+  def test_property_channel_name
+    assert_equal('default', AwsConfigDeliveryChannel.new('default').channel_name)
+    assert_equal('default', AwsConfigDeliveryChannel.new.channel_name)
+    assert_equal('NonExistentDeliveryChannel',AwsConfigDeliveryChannel.new('NonExistentDeliveryChannel').channel_name)    
+  end
+
+  def test_property_delivery_frequency_in_hours
+    assert_equal(3, AwsConfigDeliveryChannel.new('default').delivery_frequency_in_hours)
+    assert_nil(AwsConfigDeliveryChannel.new('NonExistentDeliveryChannel').delivery_frequency_in_hours)
+  end
+
+  def test_property_s3_bucket_name
+    assert_equal('my-bucket', AwsConfigDeliveryChannel.new('default').s3_bucket_name)
+    assert_nil(AwsConfigDeliveryChannel.new('NonExistentDeliveryChannel').s3_bucket_name)
+  end
+
+  def test_property_s3_key_prefix
+    assert_equal('config-logs/', AwsConfigDeliveryChannel.new('default').s3_key_prefix)
+    assert_nil(AwsConfigDeliveryChannel.new('NonExistentDeliveryChannel').s3_key_prefix)
+  end
+  
+  def test_property_sns_topic_arn
+    assert_equal('arn:aws:sns:::my-topic-name', AwsConfigDeliveryChannel.new('default').sns_topic_arn)
+    assert_nil(AwsConfigDeliveryChannel.new('NonExistentDeliveryChannel').sns_topic_arn)
+  end
+end
+
+#=============================================================================#
+#                               Test Matchers
+#=============================================================================#
+
+# None
+
+#=============================================================================#
+#                               Test Fixtures
+#=============================================================================#
+
+module AwsMDCSB
+  class Basic < AwsBackendBase
+    def describe_delivery_channels(query = {})
+      fixtures = {
+        'default' => Aws::ConfigService::Types::DescribeDeliveryChannelsResponse.new(
+          :delivery_channels => [
```

Comment:
Sometimes I see `{:foo => :bar}` and sometimes I see `{foo: :bar}`, particularly in the Rubocop-ignored test code. Shouldn't we always use JSON style for symbol keys?

Reply:
If it ignored by rubocop, that means we as a team have decided it is code that is not worthwhile to enforce style conventions.  We can open that for discussion if you'd like.